src/server/dispatcher.ts

```diff
@@ -242,18 +241,18 @@ export class Dispatcher {
     const response = await this.registry.endpoint(
       method,
       path,
-      this.parameterTypes(operation?.parameters),
+      this.parameterTypes(operation?.parameters)
     )({
       body,
-      context: this.contextRegistry.find(path),
+      context: this.contextRegistry.find(matchedPath),
```

Comment:
This is the only real change. Using `matchedPath` lets the `contextRegistry` find the given path among its entries.